Public/Sdk/Public/Deployment/definition.dsc

```diff
@@ -53,7 +53,7 @@ export interface DeployedFileWithProvenance {
 @@public
 export interface FlattenedResult {
     flattenedFiles: Map<RelativePath, DeployedFileWithProvenance>,
-    flattenedOpaques: Map<RelativePath, OpaqueDirectory>,
+    flattenedOpaques: Map<RelativePath, [OpaqueDirectory, RelativePath]>,  // Tuple of (1) an opaque directory, and (2) a path relative to that opaque directory designating a subdirectory to be added to the deployment

```

Comment:
>RelativePath [](start = 58, length = 12)

this change seems to imply that now all flattened opaques will have a designated subdirectory to be added to the deployment? what if that's not the case?

Reply:
when that's not the case, the `RelativePath` part is 
```
r`.`
```
so everything works exactly as it did before

Reply:
Yeah, saw that in a use of this. But I don't love the change, it looks like the type is being special cased for this particular problem, and now all flattened opaque will have a designated subdirectory to deploy and r`.` has to be used is all cases where this subdir thing does not apply.
I'd rather see a new type defined that represents an opaque subdirectory and make this map contain a union type of opaques or opaque subdirs

---
In reply to: [330191168](https://github.com/microsoft/BuildXL/pull/966#discussion_r330191168) [](ancestors = 330191168)